### ISSUE TITLE
Crash on download (special char related?)

### ISSUE THREAD

**Issue by digiwombat** [raulhaag/MiMangaNu#260]
When I try to download Sexual hunter Riot Chapter 20 (titled ☆☆☆☆☆☆) the app crashes outright. Assuming this is to do with a failure to handle the title as other chapters from the series work fine.

On latest build. Problem has existed since the chapter came out (1.45 maybe)

Addendum: Also happening on the latest Hantsu x Torasshu which contains only exclamation points.

Both manga are on KissManga

**Comment by digiwombat** [comment]
Looks like in HxT it wasn't the exclamation points, it was a smart quote apostraphe so instead ' it was ’. Subtle, but causing the error.

**Comment by xtj-9182** [comment]
I can confirm this. This is the Exception for Sexual Riot Hunter cha 20:

`java.lang.IllegalArgumentException: Unexpected char 0x2606 at 42 in header value: /Manga/Sexual-Hunter-Riot/Vol-004-Ch-020--☆☆☆☆☆☆?id=271966
                                                                           at com.squareup.okhttp.Headers$Builder.checkNameAndValue(Headers.java:295)
                                                                           at com.squareup.okhttp.Headers$Builder.set(Headers.java:275)
                                                                           at com.squareup.okhttp.Request$Builder.header(Request.java:185)
                                                                           at ar.rulosoft.navegadores.RefererInterceptor.intercept(RefererInterceptor.java:19)
                                                                           at com.squareup.okhttp.internal.http.HttpEngine$NetworkInterceptorChain.proceed(HttpEngine.java:867)
                                                                           at ar.rulosoft.navegadores.UserAgentInterceptor.intercept(Navegador.java:159)
                                                                           at com.squareup.okhttp.internal.http.HttpEngine$NetworkInterceptorChain.proceed(HttpEngine.java:867)
                                                                           at com.squareup.okhttp.internal.http.HttpEngine.readResponse(HttpEngine.java:748)
                                                                           at com.squareup.okhttp.Call.getResponse(Call.java:277)
                                                                           at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:234)
                                                                           at com.squareup.okhttp.Call.getResponseWithInterceptorChain(Call.java:196)
                                                                           at com.squareup.okhttp.Call.execute(Call.java:79)
                                                                           at ar.rulosoft.mimanganu.services.SingleDownload.run(SingleDownload.java:67)
                                                                           at java.lang.Thread.run(Thread.java:818)`

these three lines point to our code:
`at ar.rulosoft.navegadores.RefererInterceptor.intercept(RefererInterceptor.java:19)
at ar.rulosoft.navegadores.UserAgentInterceptor.intercept(Navegador.java:159)
at ar.rulosoft.mimanganu.services.SingleDownload.run(SingleDownload.java:67)
`

something weird is also going on with Sexual Riot Hunter cha 2 when I try to download that then I'm getting a crash? or something ... it seems weird since I don't get an exception or anything, it simply brings MiMangaNu to the background ...

**Comment by xtj-9182** [comment]
ok so it seems we only crashed cause we tried to set our referer even if it contained special characters so in my fix I simply set no referer when we encounter a referer with special characters.

However there's still something weird going on with Sexual Riot Hunter when I finish reading cha. 19 and try to go to the next chapter then I'm either brought to chapter 1 or I'm told it's the last chapter. This could be caused by the special characters in cha. 20
